test/cli.mjs

```diff
@@ -0,0 +1,90 @@
+import { spawnSync } from "child_process";
+import * as crypto from "crypto";
+import * as path from "path";
+import * as os from "os";
+
+import { default as chai, expect } from "chai";
+import chaiAsPromised from "chai-as-promised";
+chai.use(chaiAsPromised);
+
+import fs from "fs-extra";
+import rimraf from "rimraf";
+import { temporaryDirectory } from "tempy";
+
+// TODO - More test coverage
+describe("postject CLI", () => {
+  let filename;
+  let tempDir;
+  let resourceContents;
+  let resourceFilename;
+  const IS_WINDOWS = os.platform() === "win32";
+
+  beforeEach(async () => {
+    let originalFilename;
+
+    tempDir = temporaryDirectory();
+    await fs.ensureDir(tempDir);
+
+    if (IS_WINDOWS) {
+      originalFilename = "./build/test/Debug/cpp_test.exe";
+      filename = path.join(tempDir, "cpp_test.exe");
+    } else {
+      originalFilename = "./build/test/cpp_test";
+      filename = path.join(tempDir, "cpp_test");
+    }
+
+    await fs.copy(originalFilename, filename);
+
+    resourceContents = crypto.randomBytes(64).toString("hex");
+    resourceFilename = path.join(tempDir, "resource.bin");
+    await fs.writeFile(resourceFilename, resourceContents);
+  });
+
+  afterEach(() => {
+    rimraf.sync(tempDir);
+  });
+
+  it("should have help output", async () => {
+    const { status, stdout } = spawnSync("node", ["./dist/main.js", "-h"], {
+      encoding: "utf-8",
+    });
+    expect(status).to.equal(0);
+    expect(stdout).to.have.string("Usage");
+  });
+
+  it("has required arguments", async () => {
+    const { status, stderr } = spawnSync("node", ["./dist/main.js"], {
+      encoding: "utf-8",
+    });
+    expect(status).to.equal(1);
+    expect(stderr).to.have.string("required");
+  });
+
+  it("should inject a resource successfully", async () => {
+    // Before injection
+    {
+      const { status, stdout } = spawnSync(filename, { encoding: "utf-8" });
+      expect(status).to.equal(0);
+      expect(stdout).to.have.string("Hello world");
```

Comment:
```suggestion
      expect(stdout).to.equal("Hello world");
```
?

Reply:
`equal` requires an exact match (e.g., "Hello world\r\n") but the line endings differ by platform, so that gets ugly quickly. I suggest we leave it as-is for this PR, improving tests will be a follow-up PR to this to expand coverage and test surface area. For example, I want to refactor the test program to output more information that the tests can check, such as length returned by `postject_find_resource` and I'm thinking that could be dumped to `stderr` for easier parsing. During that refactoring we can look at excluding the line ending on `stdout` to make comparison the same across all platforms.